server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/SetMailboxesDestructionProcessor.java

```diff
@@ -116,9 +116,8 @@ private void destroyMailbox(Entry<MailboxId, Mailbox> entry, MailboxSession mail
             Mailbox mailbox = entry.getValue();
             preconditions(mailbox, mailboxSession);
 
-            MailboxPath mailboxPath = mailboxManager.getMailbox(mailbox.getId(), mailboxSession).getMailboxPath();
-            mailboxManager.deleteMailbox(mailboxPath, mailboxSession);
-            subscriptionManager.unsubscribe(mailboxSession, mailboxPath.getName());
+            MailboxPath deletedMailbox = mailboxManager.deleteMailbox(mailbox.getId(), mailboxSession).generateAssociatedPath();
+            subscriptionManager.unsubscribe(mailboxSession, deletedMailbox.getName());
```

Comment:
should you do the same for unsubscribe?

Reply:
Subscription is name based. Migrating it to mailboxId would be costly... I also think the spec is tricky upon rename...

```
      A server MAY validate the mailbox argument to SUBSCRIBE to verify
      that it exists.  However, it MUST NOT unilaterally remove an
      existing mailbox name from the subscription list even if a mailbox
      by that name no longer exists.

           Note: This requirement is because a server site can
           choose to routinely remove a mailbox with a well-known
           name (e.g., "system-alerts") after its contents expire,
           with the intention of recreating it when new contents
           are appropriate.
```